modules/sbt-plugin/src/main/scala/org/scalasteward/sbt/plugin/StewardPlugin.scala

```diff
@@ -56,14 +56,22 @@ object StewardPlugin extends AutoPlugin {
           Try(Credentials.forHost(sbtCredentials, url.getHost)).toOption.flatten
             .map(c => Resolver.Credentials(c.userName, c.passwd))
 
+        def getHeaders(name: String): List[Resolver.Header] =
+          for {
+            (repositoryId, authentication) <-
+              csrConfiguration.value.authenticationByRepositoryId.toList
+            if repositoryId == name
+            (headerKey, headerValue) <- authentication.headers
+          } yield Resolver.Header(headerKey, headerValue)
```

Comment:
I think you should not only take the headers from the `csrConfiguration` but also username and password.
Setting these via `csrConfiguration` is not wrong. And I would prefer `csrConfiguration` over `credentials` if present, since it is more specific/explicit.

Funny enough, I did not see this pr and coded it myself that way. Maybe you can have a [look](https://github.com/987Nabil/scala-steward/blob/import-csr-http-headers/modules/sbt-plugin/src/main/scala/org/scalasteward/sbt/plugin/StewardPlugin.scala) ;)

Reply:
I forgot the @markvandertol :)

Reply:
I am going to merge as-is since markvandertol seems busy now.
@987Nabil You can submit a PR for that change 😉

Reply:
Sorry for not replying to this earlier. It indeed looks more consistent, but I'm not entirely sure how large of a breaking change would be, e.g. for people who disabled _coursier_ in their projects. Retrieving the credentials old way, but looking for headers in `csrConfiguration` shouldn't be breaking, so I propose to do that in a different PR so the impact can better be assessed.

Reply:
We can do this in another PR for sure. But I think, deactivating coursier and then relying on `csrConfiguration` where csr = coursier and the [description](https://github.com/sbt/sbt/blob/1.7.x/main/src/main/scala/sbt/Keys.scala#L436) in sbt also explicitly mentions coursier, seems to be odd.